SSL tests fail on Ubuntu 24.10 with openssl 3.3.1

Hello! I have already created similar issue once (#1062). But the system was specific. Now I run tests on Ubuntu 24.10 with new openssl 3.3.1 installed:
```
test@ubuntu2410:~$ openssl --version
OpenSSL 3.3.1 4 Jun 2024 (Library: OpenSSL 3.3.1 4 Jun 2024)
```
run test:
```
pytest test/test_ssl.py
```
result:
```
FAILED test/test_ssl.py::test_server_ssl_verify - psycopg.errors.ConnectionTimeout: connection timeout expired
FAILED test/test_ssl.py::test_server_ssl_auth - psycopg.errors.ConnectionTimeout: connection timeout expired
FAILED test/test_ssl.py::test_client_ssl_set_enable_disable - subprocess.CalledProcessError: Command '['psql', 'port=10208 host=localhost sslmode=verify-full sslrootcert=/tmp/pytest-of-test/pytest-3/certs/TestCA1/ca.crt dbname=p0 user=postgres port=10208 connect_timeout=3 client_encoding=UTF8', '-c', 'select...
FAILED test/test_ssl.py::test_client_ssl_set_change_ca - subprocess.CalledProcessError: Command '['psql', 'port=10209 host=localhost sslmode=verify-full sslrootcert=/tmp/pytest-of-test/pytest-3/certs/TestCA1/ca.crt dbname=p0 user=postgres port=10209 connect_timeout=3 client_encoding=UTF8', '-c', 'select...
FAILED test/test_ssl.py::test_client_ssl_sighup_enable_disable - subprocess.CalledProcessError: Command '['psql', 'port=10210 host=localhost sslmode=verify-full sslrootcert=/tmp/pytest-of-test/pytest-3/certs/TestCA1/ca.crt dbname=p0 user=postgres port=10210 connect_timeout=3 client_encoding=UTF8', '-c', 'select...
FAILED test/test_ssl.py::test_client_ssl_sighup_change_ca - subprocess.CalledProcessError: Command '['psql', 'port=10211 host=localhost sslmode=verify-full sslrootcert=/tmp/pytest-of-test/pytest-3/certs/TestCA1/ca.crt dbname=p0 user=postgres port=10211 connect_timeout=3 client_encoding=UTF8', '-c', 'select...
FAILED test/test_ssl.py::test_client_ssl_reload_enable_disable - subprocess.CalledProcessError: Command '['psql', 'port=10212 host=localhost sslmode=verify-full sslrootcert=/tmp/pytest-of-test/pytest-3/certs/TestCA1/ca.crt dbname=p0 user=postgres port=10212 connect_timeout=3 client_encoding=UTF8', '-c', 'select...
FAILED test/test_ssl.py::test_client_ssl_reload_change_ca - subprocess.CalledProcessError: Command '['psql', 'port=10213 host=localhost sslmode=verify-full sslrootcert=/tmp/pytest-of-test/pytest-3/certs/TestCA1/ca.crt dbname=p0 user=postgres port=10213 connect_timeout=3 client_encoding=UTF8', '-c', 'select...
FAILED test/test_ssl.py::test_client_ssl_auth - subprocess.CalledProcessError: Command '['psql', 'port=10214 host=localhost sslmode=verify-full user=bouncer sslrootcert=/tmp/pytest-of-test/pytest-3/certs/TestCA1/ca.crt sslkey=/tmp/pytest-of-test/pytest-3/certs/TestCA1/sites/02-bouncer.key sslce...
FAILED test/test_ssl.py::test_client_ssl_scram - subprocess.CalledProcessError: Command '['psql', 'port=10215 host=localhost user=bouncer password=zzzz sslmode=verify-full sslrootcert=/tmp/pytest-of-test/pytest-3/certs/TestCA1/ca.crt dbname=p0 port=10215 connect_timeout=3 client_encoding=UTF8', ...
FAILED test/test_ssl.py::test_ssl_replication - subprocess.CalledProcessError: Command '['psql', 'port=10216 host=localhost dbname=p7a replication=database user=postgres application_name=abc sslmode=verify-full sslrootcert=/tmp/pytest-of-test/pytest-3/certs/TestCA1/ca.crt port=10216 connect_tim...
```
In logfile I get messages like these:
```
psql: error: connection to server at "localhost" (127.0.0.1), port 10213 failed: SSL error: certificate verify failed
```
or
```
WARNING TLS handshake error: handshake failed: error:0A000418:SSL routines::tlsv1 alert unknown ca
```
Also test/test_auth.py::test_client_hba_cert failes the same way:
```
FAILED test/test_auth.py::test_client_hba_cert - subprocess.CalledProcessError: Command '['psql', 'port=10223 dbname=p0x host=localhost user=someuser sslmode=verify-full sslkey=/tmp/pytest-of-test/pytest-5/certs/TestCA1/sites/04-pgbouncer.acme.org.key sslcert=/tmp/pytest-of-test/pytest-5/certs/T...
```
```
WARNING TLS handshake error: handshake failed: error:0A000418:SSL routines::tlsv1 alert unknown ca
```

--- a/test/ssl/newca.sh
+++ b/test/ssl/newca.sh
@@ -30,7 +30,7 @@ days=10240
 run openssl ecparam -name prime256v1 -genkey -out "$name/ca.key"
 
 # self-signed cert
-run_req -new -x509 -days $days -key "$name/ca.key" -out "$name/ca.crt" -- "$@"
+run_req -new -x509 -days $days -key "$name/ca.key" -out "$name/ca.crt" -addext basicConstraints=critical,CA:TRUE,pathlen:1 -- "$@"
 
 
 cat > "$name"/config.ini <<EOF

--- a/test/test_ssl.py
+++ b/test/test_ssl.py
@@ -9,11 +9,6 @@ from .utils import MACOS, PG_MAJOR_VERSION, TEST_DIR, TLS_SUPPORT, WINDOWS, Boun
 if not TLS_SUPPORT:
     pytest.skip(allow_module_level=True)
 
-# Windows and MacOS TLS tests are currently broken for some strange reason.
-# Make CI pass for now by ignoring these tests.
-if WINDOWS or MACOS:
-    pytest.skip(allow_module_level=True)
-
 # XXX: These test use psql to connect using sslmode=verify-full instead of
 # using psycopg. The reason for this is that psycopg has a bug on Apple
 # silicon when enabling SSL: https://github.com/psycopg/psycopg/discussions/270

